src/Analyser/MutatingScope.php

```diff
@@ -1400,6 +1400,11 @@ private function resolveType(Expr $node): Type
 				return new ErrorType();
 			}
 
+			if ($leftType->isNonEmptyString()->yes() && !$leftType->isNumericString()->yes() ||
```

Comment:
1) `||` vs. `&&` operator priority needs to be disambiguated with parentheses
2) Why `isNonEmptyString` and not `isString`?